keyboards/planck/rev7/keymaps/default/keymap.c

```diff
@@ -165,7 +221,61 @@ layer_state_t layer_state_set_user(layer_state_t state) {
     return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
 }
 
+/* clang-format off */
+float melody[8][2][2] = {
+    {{440.0f, 8}, {440.0f, 24}},
+    {{440.0f, 8}, {440.0f, 24}},
+    {{440.0f, 8}, {440.0f, 24}},
+    {{440.0f, 8}, {440.0f, 24}},
+    {{440.0f, 8}, {440.0f, 24}},
+    {{440.0f, 8}, {440.0f, 24}},
+    {{440.0f, 8}, {440.0f, 24}},
+    {{440.0f, 8}, {440.0f, 24}},
+};
+/* clang-format on */
+
+#define JUST_MINOR_THIRD 1.2
+#define JUST_MAJOR_THIRD 1.25
+#define JUST_PERFECT_FOURTH 1.33333333
+#define JUST_TRITONE 1.42222222
+#define JUST_PERFECT_FIFTH 1.33333333
+
+#define ET12_MINOR_SECOND 1.059463
+#define ET12_MAJOR_SECOND 1.122462
+#define ET12_MINOR_THIRD 1.189207
+#define ET12_MAJOR_THIRD 1.259921
+#define ET12_PERFECT_FOURTH 1.33484
+#define ET12_TRITONE 1.414214
+#define ET12_PERFECT_FIFTH 1.498307
+
+deferred_token tokens[8];
+
+uint32_t reset_note(uint32_t trigger_time, void *note) {
+    *(float*)note = 440.0f;
+    return 0;
+}
+
+bool play_encoder_melody(uint8_t index, bool clockwise) {
+    cancel_deferred_exec(tokens[index]);
+    if (clockwise) {
+        melody[index][1][0] = melody[index][1][0] * ET12_MINOR_SECOND;
+        melody[index][0][0] = melody[index][1][0] / ET12_PERFECT_FIFTH;
+        audio_play_melody(&melody[index], 2, false);
+    } else {
+        melody[index][1][0] = melody[index][1][0] / ET12_MINOR_SECOND;
+        melody[index][0][0] = melody[index][1][0] * ET12_TRITONE;
+        audio_play_melody(&melody[index], 2, false);
+    }
+    tokens[index] = defer_exec(1000, reset_note, &melody[index][1][0]);
+    return false;
+}
+
```

Comment:
Can we revert the positioning on this, just providing a forward declaration of `play_encoder_melody` for use within `process_record_user`? Reduces the amount of code churn.

Reply:
@tzarc Thank you for your review. I did change a bit. Please check commits.